src/window_identifier/wayland.rs

```diff
@@ -1,40 +1,79 @@
 use std::fmt;
 
+use wayland_backend::sys::client::Backend;
 use wayland_client::{
-    protocol::{__interfaces::WL_SURFACE_INTERFACE, wl_surface::WlSurface},
+    protocol::{wl_registry, wl_surface::WlSurface},
     Proxy, QueueHandle,
 };
-use wayland_protocols::xdg::foreign::zv2::client::{
-    zxdg_exported_v2::{Event, ZxdgExportedV2},
-    zxdg_exporter_v2::ZxdgExporterV2,
+use wayland_protocols::xdg::foreign::{
+    zv1::client::{
+        zxdg_exported_v1::{self, ZxdgExportedV1},
+        zxdg_exporter_v1::ZxdgExporterV1,
+    },
+    zv2::client::{
+        zxdg_exported_v2::{self, ZxdgExportedV2},
+        zxdg_exporter_v2::ZxdgExporterV2,
+    },
 };
 
 use super::WindowIdentifierType;
 
+// Supported versions.
+const ZXDG_EXPORTER_V1: u32 = 1;
+const ZXDG_EXPORTER_V2: u32 = 1;
+
+#[derive(Debug)]
 pub struct WaylandWindowIdentifier {
-    exported: ZxdgExportedV2,
+    exported_v1: Option<ZxdgExportedV1>,
+    exported_v2: Option<ZxdgExportedV2>,
     type_: WindowIdentifierType,
 }
```

Comment:
Do we need a `Drop` implementation to destroy the inner `exported` handles?

Do note that destroying exported handles will revoke any imports performed on other clients and the `handle` string will become inert.

Reply:
So how it should be handled? Should we expose a method for the user to drop the handle?

Reply:
gtk3 and gtk4 appear to destroy their exported handles. So yes `Drop` should destroy the exported protocol object.

Reply:
Yes, I wanted to be consistent with gtk's behaviour. Although, the issue with gtk is it errors out when you try to re-export a handle for an already exported surface instead of returning the same handle. Which make the WindowIdentifier not very nice with gtk, that is why I made it a kind of reference counted from ashpd's side and only unexport when the last usage is dropped.

I am thinking of going directly through the wayland-rs crates in the future and avoid using gtk's provided methods.